documentation/book/con-kafka-listeners.adoc

```diff
@@ -148,7 +151,89 @@ listeners:
 
 For more information on using node ports to access Kafka, see xref:proc-accessing-kafka-using-nodeports-{context}[].
 
-.Customizing advertised addresses on external listeners
+ifdef::Kubernetes[]
+=== Exposing Kafka using {KubernetesName} Ingress
+
+An external listener of type `ingress` exposes Kafka by using {KubernetesName} `Ingress` and the {NginxIngressController}.
+A dedicated `Ingress` resource is created for every Kafka broker pod.
+An additional `Ingress` resource is created to serve as a Kafka bootstrap address.
+Kafka clients can use these `Ingress` resources to connect to Kafka on port 443.
+
+NOTE: External listeners using `Ingress` have been currently tested only with the {NginxIngressController}.
+
+{ProductName} is using the TLS passthrough feature of the {NginxIngressController}.
+Make sure TLS passthrough is enabled in your {NginxIngressController} deployment.
+For more information abotu enabling TLS passthrough see {NginxIngressControllerTLSPassthrough}.
+Because it is using the TLS passthrough functionality, TLS encryption will be always used when exposing Kafka using `Ingress`.
+
+The Ingress controller does not assign any hostnames automatically.
+You have to specify the hostnames which should be used by the bootstrap and per-broker services in the `configuration` section.
+You also have to make sure that the hostnames resolve to the Ingress endpoints.
+{ProductName} will not perform any validation that the requested hosts are available and properly routed to the Ingress endpoints.
```

Comment:
I suppose we _could_ do this, if we thought it was worth the effort.

Reply:
That would be quite complicated ... we would need to setup our own backend, setup a test ingress for it and check that we can reach it. And that alone might still not work if there are some network restrictions, DNS limitations etc. So I would probably say that we could do it, but it would be very complicated.